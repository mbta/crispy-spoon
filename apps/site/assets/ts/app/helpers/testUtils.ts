/* eslint-disable */
import toJson, { Json } from "enzyme-to-json";
import { ReactWrapper } from "enzyme";
import {
  IndexedRouteStop,
  StopTree
} from "../../schedule/components/__schedule";

export const testRouteStopListFromStopTree = (
  tree: StopTree
): IndexedRouteStop[] =>
  Object.values(tree.byId).map((node, index) => ({
    ...node.value,
    routeIndex: index
  }));

export const createReactRoot = (): void => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';
};

export const enzymeToJsonWithoutProps = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false,
    map: info => {
      if (!info.props) {
        return info;
      }

      // @ts-ignore is a <style></style> declaration
      if (info.node.type === "style") {
        return info;
      }
      // @ts-ignore is an element in a <noscript></noscript> declaration
      if (info.node.nodeType == undefined) {
        // @ts-ignore
        return info;
      }
      //@ts-ignore is a React component (not HTML element)
      if (info.node.nodeType.toString().includes("function")) {
        // @ts-ignore
        info.props = undefined;
      }

      return info;
    }
  });

export const toJsonEnzyme = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false
  });
