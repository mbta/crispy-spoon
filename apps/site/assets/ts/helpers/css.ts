import {
  isABusRoute,
  isACommuterRailRoute,
  isAGreenLineRoute,
  isASilverLineRoute
} from "../models/route";
import { Route } from "../__v3api";

export const routeToCSSClass = (route: string): string =>
  route.toLowerCase().replace(/[ _]/g, "-");

export const busClass = (route: Route): string =>
  isABusRoute(route) && !isASilverLineRoute(route) ? "bus-route-sign" : "";

export const routeToModeName = (route: Route): string => {
  if (isACommuterRailRoute(route)) return "commuter-rail";
  if (route.type === 4) return "ferry";
  if (route.id === "Red" || route.id === "Mattapan") return "red-line";
  if (route.id === "Orange") return "orange-line";
  if (route.id === "Blue") return "blue-line";
  if (isAGreenLineRoute(route)) return "green-line";
  if (isASilverLineRoute(route)) return "silver-line";
  if (isABusRoute(route)) return "bus";
  return "unknown";
};

export const routeBgClass = (route: Route): string =>
  `u-bg--${routeToModeName(route)}`;
