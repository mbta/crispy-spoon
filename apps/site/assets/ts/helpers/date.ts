// Returns a date in the form of "August 7, 2019"
export const formattedDate = (unformatted: string): string => {
  const [year, month, day] = unformatted
    .split(/-/)
    .map(part => Number.parseInt(part, 10));

  // Remember that months in JS are 0-indexed for some reason, hence "month - 1".
  const parsedDate = new Date(year, month - 1, day);
  return parsedDate.toLocaleDateString("en-US", {
    year: "numeric",
    month: "long",
    day: "numeric",
    timeZone: "UTC"
  });
};

// Returns a date in the form of "Aug 7"
export const shortDate = (date: Date): string =>
  date.toLocaleDateString("en-US", {
    month: "short",
    day: "numeric",
    timeZone: "UTC"
  });

export default formattedDate;
