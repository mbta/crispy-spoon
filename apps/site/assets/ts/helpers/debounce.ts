// Debounce is meant to be used with event listeners. Use it to delay a call
// until after a burst of events, such as keyboard input.
//
//   usage:
//
//     const callback = (ev: Event) => {}
//     document.addEventListener("keyup", debounce(callback, 250))
//

const debounce = (callback: Function, delay: number): EventListener => {
  let timer: number | undefined;

  return (...args) => {
    window.clearTimeout(timer);
    timer = window.setTimeout(() => callback(...args), delay);
  };
};

export const debouncePromise = (
  callback: Function,
  delay: number
): Function => {
  let timerId: number | undefined;

  return function debounced(...args: unknown[]) {
    if (timerId) {
      window.clearTimeout(timerId);
    }

    return new Promise(resolve => {
      timerId = window.setTimeout(() => resolve(callback(...args)), delay);
    });
  };
};

export default debounce;
