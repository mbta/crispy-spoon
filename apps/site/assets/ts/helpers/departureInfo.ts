import { concat, difference, keyBy, keys, map, sortBy } from "lodash";
import { PredictionWithTimestamp } from "../models/perdictions";
import { ScheduleWithTimestamp } from "../models/schedules";
import { isCancelled, isDelayed } from "./prediction-helpers";
import { DepartureInfo } from "../models/departureInfo";

const departureInfoToTime = (departureInfo: DepartureInfo): Date => {
  // If there isn't a prediction there should be a schedule
  return departureInfo.prediction
    ? departureInfo.prediction.time
    : departureInfo.schedule!.time;
};

const mergeIntoDepartureInfo = (
  schedules: ScheduleWithTimestamp[],
  predictions: PredictionWithTimestamp[]
): DepartureInfo[] => {
  const schedulesByTripId = keyBy(schedules, sch => sch.trip.id);
  const predictionsByTripId = keyBy(predictions, prd => prd.trip.id);

  const scheduleTripIds = keys(schedulesByTripId);
  const predictionTripIds = keys(predictionsByTripId);

  // All the schedules with and without predictions
  const scheduleDirectionInfo = map(
    schedulesByTripId,
    (schedule, tripId): DepartureInfo => {
      // prediction could be undefined
      const prediction = predictionsByTripId[tripId];
      return {
        prediction: prediction,
        schedule: schedule,
        isCancelled: isCancelled(prediction),
        isDelayed: isDelayed(prediction, schedule)
      };
    }
  );

  const unmatchedPredictionTripIds = difference(
    predictionTripIds,
    scheduleTripIds
  );
  // All the predictions with no schedules
  const predictionDirectionInfo = map(
    unmatchedPredictionTripIds,
    (tripId: string): DepartureInfo => {
      const prediction = predictionsByTripId[tripId];
      return {
        prediction: prediction,
        isCancelled: isCancelled(prediction)
      } as DepartureInfo;
    }
  );

  const departureInfos = concat(scheduleDirectionInfo, predictionDirectionInfo);
  const sortedDepartureInfo = sortBy(departureInfos, di => {
    // prioritize sorting by predictions
    if (di.prediction) {
      return di.prediction.time;
    }
    return di.schedule?.time;
  });

  return sortedDepartureInfo;
};

export { mergeIntoDepartureInfo, departureInfoToTime };
