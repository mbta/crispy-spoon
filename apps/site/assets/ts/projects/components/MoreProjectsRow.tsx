import React, { ReactElement } from "react";
import { isASilverLineRoute } from "../../models/route";
import formattedDate from "../../helpers/date";
import { Teaser as Project, Route } from "../../__cms";
import RouteIcon from "./RouteIcon";

interface Props extends Project {
  placeholderImageUrl: string;
}

const busTags = (routes: Route[]): string[] => {
  if (routes.find(route => route.mode === "bus")) {
    return ["bus"];
  }
  return [];
};

const commuterRailTags = (routes: Route[]): string[] => {
  if (routes.find(route => route.mode === "commuter_rail")) {
    return ["commuterRail"];
  }
  return [];
};

const ferryTags = (routes: Route[]): string[] => {
  if (routes.find(route => route.mode === "ferry")) {
    return ["ferry"];
  }
  return [];
};

const subwayTags = (routes: Route[]): string[] => {
  let result = [] as string[];

  if (routes.find(route => route.id === "subway")) {
    result.push("subway");
  }

  if (routes.find(route => route.id === "Blue")) {
    result.push("blue");
  }

  if (routes.find(route => route.id === "Green")) {
    result.push("green");
  }

  const greenLineBranches = routes.filter(route => route.id.match(/^Green-/));
  result = result
    .concat(greenLineBranches.map(route => route.id.toLowerCase()))
    .sort();

  if (routes.find(route => route.id === "Mattapan")) {
    result.push("mattapan");
  }

  if (routes.find(route => route.id === "Orange")) {
    result.push("orange");
  }

  if (routes.find(route => route.id === "Red")) {
    result.push("red");
  }

  if (routes.find(route => isASilverLineRoute(route.id))) {
    result.push("silver");
  }

  return result;
};

export const routesToTags = (routes: Route[]): string[] =>
  busTags(routes)
    .concat(subwayTags(routes))
    .concat(ferryTags(routes))
    .concat(commuterRailTags(routes));

const MoreProjectsRow = ({
  image,
  path,
  title,
  routes,
  date,
  placeholderImageUrl,
  status
}: Props): ReactElement<HTMLElement> => {
  const normalizedStatus = status ? status.toLowerCase().replace(/ /, "-") : "";
  const statusTextClassName = status
    ? `m-more-projects-table__status-text--${normalizedStatus}`
    : "";
  const statusIconClassName = status
    ? `m-more-projects-table__status-icon--${normalizedStatus}`
    : "";

  return (
    <tr className="m-more-projects-table__tr">
      <td className="m-more-projects-table__td m-more-projects-table__td-project">
        {(image && (
          <img
            src={image.url}
            alt={image.alt}
            className="hidden-xs-down m-more-projects-table__thumbnail"
          />
        )) || (
          <img
            src={placeholderImageUrl}
            alt="MBTA logo"
            className="hidden-xs-down m-more-projects-table__thumbnail"
          />
        )}

        <div className="m-more-projects-table__wrapper">
          <a href={path}>
            <h3 className="m-more-projects-table__title">{title}</h3>
          </a>

          <div className="m-more-projects-table__updated-and-status hidden-lg-up">
            <div className="m-more-projects-table__last-updated">
              <span className="m-more-projects-table__internal-header">
                Last updated
              </span>
              <br />
              {formattedDate(date)}
              <br />
              {routesToTags(routes).map(tag => (
                <RouteIcon
                  key={tag}
                  tag={tag}
                  extraClasses="m-more-projects-table__route-icon hidden-lg-up"
                />
              ))}
            </div>

            {status && (
              <div className="m-more-projects-table__status">
                <span className="m-more-projects-table__internal-header">
                  Status
                </span>
                <br />
                <i
                  className={`fa m-more-projects-table__status-icon ${statusIconClassName}`}
                />{" "}
                <span
                  className={`m-more-projects-table__status-text ${statusTextClassName}`}
                >
                  {status}
                </span>
              </div>
            )}
          </div>

          {routesToTags(routes).map(tag => (
            <RouteIcon
              key={tag}
              tag={tag}
              extraClasses="m-more-projects-table__route-icon hidden-md-down"
            />
          ))}
        </div>
      </td>

      <td className="m-more-projects-table__td m-more-projects-table__td-last-updated hidden-md-down">
        {formattedDate(date)}
      </td>

      <td className="m-more-projects-table__td m-more-projects-table__td-status hidden-md-down">
        <span className="m-more-projects-table__status-wrapper">
          {status && (
            <>
              <i
                className={`fa m-more-projects-table__status-icon ${statusIconClassName}`}
              />{" "}
              <span
                className={`m-more-projects-table__status-text ${statusTextClassName}`}
              >
                {status}
              </span>
            </>
          )}
        </span>
      </td>
    </tr>
  );
};

export default MoreProjectsRow;
