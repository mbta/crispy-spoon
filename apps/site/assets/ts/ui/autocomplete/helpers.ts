import { omit } from "lodash";
import { OnSubmitParams } from "@algolia/autocomplete-core";
import { OnStateChangeProps } from "@algolia/autocomplete-js";
import {
  ContentItem,
  Item,
  LocationItem,
  PopularItem,
  RouteItem,
  StopItem
} from "./__autocomplete";
import { isLGDown } from "../../helpers/media-breakpoints";

export function isStopItem(x: Item): x is StopItem {
  return Object.keys(x).includes("stop");
}

export function isRouteItem(x: Item): x is RouteItem {
  return Object.keys(x).includes("route");
}

export function isContentItem(x: Item): x is ContentItem {
  return Object.keys(x).includes("_content_type");
}

export const getTitleAttribute = (item: Item): string[] => {
  if (isStopItem(item)) {
    return ["stop", "name"];
  }
  if (isRouteItem(item)) {
    return ["route", "name"];
  }

  // @ts-ignore
  // eslint-disable-next-line no-underscore-dangle
  return Object.keys(item._highlightResult);
};

const navStateChange: (props: OnStateChangeProps<Item>) => void = ({
  state
}) => {
  // grey out the page and disable scrolling when search is open
  if (isLGDown()) {
    if (state.isOpen) {
      document.documentElement.dataset.navOpen = "true";
    } else {
      delete document.documentElement.dataset.navOpen;
    }
  }
};

export const STATE_CHANGE_HANDLERS: Record<
  string,
  (props: OnStateChangeProps<Item>) => void
> = {
  nav: navStateChange
};

export const SUBMIT_HANDLERS: Record<
  string,
  (props: OnSubmitParams<Item>) => void
> = {
  to_search_page({ state }) {
    window.Turbolinks.visit(`/search?query=${state.query}`);
  }
};

export type WithUrls<T> = T & { urls: Record<string, string> };

// The backend returns all possible URLs, use urlType to get the desired one
export const itemWithUrl = (
  initialItem: WithUrls<LocationItem | PopularItem>,
  urlType: string
): Omit<typeof initialItem, "urls"> => {
  const item = omit(initialItem, "urls");
  item.url = initialItem.urls[urlType];
  return item;
};
