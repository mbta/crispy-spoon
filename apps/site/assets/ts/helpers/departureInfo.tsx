import React from "react";
import {
  chain,
  concat,
  difference,
  find,
  keyBy,
  keys,
  map,
  sortBy
} from "lodash";
import { PredictionWithTimestamp } from "../models/predictions";
import { ScheduleWithTimestamp } from "../models/schedules";
import { isCancelled, isDelayed, isSkipped } from "./prediction-helpers";
import { DepartureInfo } from "../models/departureInfo";
import {
  isABusRoute,
  isACommuterRailRoute,
  isSubwayRoute
} from "../models/route";
import { DirectionId, Route } from "../__v3api";
import DisplayTime from "../stop/components/DisplayTime";
import { getInfoKey } from "../stop/models/displayTimeConfig";
import { RoutePatternWithPolyline } from "../models/route-patterns";

export const SUBWAY = "subway";
export const BUS = "bus";
export const COMMUTER_RAIL = "commuter_rail";
export const FERRY = "ferry";

const toRouteMode = (
  route: Route
): typeof SUBWAY | typeof BUS | typeof COMMUTER_RAIL | typeof FERRY => {
  if (isSubwayRoute(route)) {
    return SUBWAY;
  }
  if (isABusRoute(route)) {
    return BUS;
  }
  if (isACommuterRailRoute(route)) {
    return COMMUTER_RAIL;
  }
  return FERRY;
};

const departureInfoToTime = (departureInfo: DepartureInfo): Date => {
  // If there isn't a prediction there should be a schedule
  return departureInfo.prediction && departureInfo.prediction.time
    ? departureInfo.prediction.time
    : departureInfo.schedule!.time;
};

const getNextUnCancelledDeparture = (
  departureInfos: DepartureInfo[]
): DepartureInfo | undefined => {
  return find(departureInfos, (di: DepartureInfo) => !di.isCancelled);
};

const displayInfoContainsPrediction = (
  departureInfo: DepartureInfo
): boolean => {
  return !!departureInfo.prediction;
};

const mergeIntoDepartureInfo = (
  schedules: ScheduleWithTimestamp[],
  predictions: PredictionWithTimestamp[]
): DepartureInfo[] => {
  const schedulesByTripId = keyBy(schedules, "trip.id");
  const predictionsByTripId = keyBy(predictions, "trip.id");
  const scheduleTripIds = keys(schedulesByTripId);
  const predictionTripIds = keys(predictionsByTripId);

  // All the schedules with and without predictions
  const scheduleDirectionInfo = map(
    schedulesByTripId,
    (schedule, tripId): DepartureInfo => {
      // prediction could be undefined
      const prediction = predictionsByTripId[tripId];
      return {
        prediction,
        schedule,
        route: schedule.route,
        trip: schedule.trip,
        isCancelled: isCancelled(prediction),
        isSkipped: isSkipped(prediction),
        isDelayed: isDelayed(prediction, schedule),
        routeMode: toRouteMode(schedule.route)
      };
    }
  );

  const unmatchedPredictionTripIds = difference(
    predictionTripIds,
    scheduleTripIds
  );
  // All the predictions with no schedules
  const predictionDirectionInfo = map(
    unmatchedPredictionTripIds,
    (tripId: string): DepartureInfo => {
      const prediction = predictionsByTripId[tripId];
      return {
        prediction,
        route: prediction.route,
        trip: prediction.trip,
        isCancelled: isCancelled(prediction),
        isSkipped: isSkipped(prediction),
        routeMode: toRouteMode(prediction.route)
      } as DepartureInfo;
    }
  );

  const departureInfos = concat(scheduleDirectionInfo, predictionDirectionInfo);
  const sortedDepartureInfo = sortBy(departureInfos, di => {
    // prioritize sorting by predictions
    if (di.prediction && di.prediction.time) {
      return di.prediction.time;
    }
    return di.schedule?.time;
  });

  return sortedDepartureInfo;
};

const isAtDestination = (
  stopName: string,
  route: Route,
  directionId: DirectionId
): boolean => {
  const destination = route.direction_destinations[directionId];
  if (destination) {
    return destination.includes(stopName) && !destination.includes(" or ");
  }
  return false;
};

const isCommuterRail = (departureInfo: DepartureInfo): boolean => {
  return departureInfo?.routeMode === "commuter_rail";
};

const DefaultWrapper: React.FunctionComponent<unknown> = ({ children }) => (
  <li>{children}</li>
);
/**
 * From a list of `DepartureInfo[]`, generate a list of `<DisplayTime />`.
 * Each `<DisplayTime />` will recieve the `isCR` and `targetDate` props.
 *
 * This function can optionally
 * - wrap each `<DisplayTime />` in a specified wrapper element, `WrapperEl`
 * - truncate the list to a desired `listLength` value
 */
const departuresListFromInfos = (
  departureInfos: DepartureInfo[],
  isCR: boolean,
  targetDate?: Date,
  listLength?: number,
  omitCancelledAndSkipped = false,
  WrapperEl: typeof DefaultWrapper = DefaultWrapper
): React.ReactElement[] => {
  // optional cutoff time, before which we won't show schedules.
  // just used with subway for now.
  const predictionTimeCutoff = chain(departureInfos)
    .filter(d => d.routeMode === SUBWAY)
    .maxBy("prediction.time")
    .value()?.prediction!.time;

  return chain(departureInfos)
    .reject(
      departure =>
        omitCancelledAndSkipped &&
        (!!departure.isCancelled || !!departure.isSkipped)
    )
    .omitBy(
      ({ prediction, schedule }) =>
        // omit schedule-only departures that are before latest prediction time
        predictionTimeCutoff &&
        !prediction &&
        schedule &&
        schedule.time <= predictionTimeCutoff
    )
    .map(d => (
      <WrapperEl key={getInfoKey(d)}>
        <DisplayTime departure={d} isCR={isCR} targetDate={targetDate} />
      </WrapperEl>
    ))
    .slice(0, listLength)
    .value();
};

const departureInfoInRoutePatterns = (
  departureInfo: DepartureInfo,
  routePatterns: RoutePatternWithPolyline[]
): boolean => {
  const isMatch = !!routePatterns.find(
    rp => rp.id === departureInfo.trip.route_pattern_id
  );
  // This second case is needed for the edge case where the route pattern is
  // canonical but the schedules/predictions are not serving canonical route
  // patterns (e.g. due to shuttles or other disruptions)
  const isSimilar = !!routePatterns.find(
    rp =>
      rp.headsign === departureInfo.trip.headsign &&
      rp.route_id === departureInfo.route.id
  );
  return isMatch || isSimilar;
};

export {
  mergeIntoDepartureInfo,
  departureInfoToTime,
  displayInfoContainsPrediction,
  getNextUnCancelledDeparture,
  isAtDestination,
  isCommuterRail,
  departuresListFromInfos,
  departureInfoInRoutePatterns
};
