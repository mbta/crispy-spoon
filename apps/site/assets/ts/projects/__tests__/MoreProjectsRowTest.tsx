import React from "react";
import renderer from "react-test-renderer";
import { createReactRoot } from "../../app/helpers/testUtils";
import MoreProjectsRow, { routesToTags } from "../components/MoreProjectsRow";
import { shallow } from "enzyme";

const id = 12345;

const image = {
  url: "http://cloudfront.com/mbta_pic.gif",
  alt: "Picture of the new Mauve Line train"
};

const path = "http://www.mbta.com/new_mauve_line";

const title = "New Mauve Line trains";

const text =
  "The Mauve Line connects Lynn to Braintree with a stop in Worcester.";

const date = "2018-05-01";

const status = "Contract awarded";

const routes = [
  { mode: "bus", id: "123", group: "route" },
  { mode: "commuter_rail", id: "CR-Anyplace", group: "line" },
  { mode: "ferry", id: "Boat-Somewhere", group: "line" },
  { mode: "subway", id: "subway", group: "mode" },
  { mode: "subway", id: "Blue", group: "line" },
  { mode: "subway", id: "Green", group: "line" },
  { mode: "subway", id: "Green-B", group: "line" },
  { mode: "subway", id: "Green-C", group: "line" },
  { mode: "subway", id: "Green-D", group: "line" },
  { mode: "subway", id: "Green-E", group: "line" },
  { mode: "subway", id: "Mattapan", group: "line" },
  { mode: "subway", id: "Orange", group: "line" },
  { mode: "subway", id: "Red", group: "line" },
  { mode: "bus", id: "746", group: "line" }
];

const placeholderImageUrl = "https://www.example.com/someimage.gif";

it("renders with route icons and status", () => {
  createReactRoot();
  const tree = renderer
    .create(
      <MoreProjectsRow
        image={image}
        path={path}
        title={title}
        text={text}
        date={date}
        id={id}
        status={status}
        routes={routes}
        placeholderImageUrl={placeholderImageUrl}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a placeholder image for a project with no teaser image", () => {
  const wrapper = shallow(
    <MoreProjectsRow
      image={null}
      path={path}
      title={title}
      text={text}
      date={date}
      id={id}
      status={null}
      routes={routes}
      placeholderImageUrl={placeholderImageUrl}
    />
  );

  const actualSrc = wrapper.find("img").prop("src");
  expect(actualSrc).toEqual(placeholderImageUrl);
});

it("returns an empty list of tags if no routes given", () => {
  expect(routesToTags([])).toEqual([]);
});
