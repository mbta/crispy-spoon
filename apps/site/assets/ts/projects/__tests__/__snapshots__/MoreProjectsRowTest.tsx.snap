// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with route icons and status 1`] = `
<tr
  className="m-more-projects-table__tr"
>
  <td
    className="m-more-projects-table__td m-more-projects-table__td-project"
  >
    <img
      alt="Picture of the new Mauve Line train"
      className="hidden-xs-down m-more-projects-table__thumbnail"
      src="http://cloudfront.com/mbta_pic.gif"
    />
    <div
      className="m-more-projects-table__wrapper"
    >
      <a
        href="http://www.mbta.com/new_mauve_line"
      >
        <h3
          className="m-more-projects-table__title"
        >
          New Mauve Line trains
        </h3>
      </a>
      <div
        className="m-more-projects-table__updated-and-status hidden-lg-up"
      >
        <div
          className="m-more-projects-table__last-updated"
        >
          <span
            className="m-more-projects-table__internal-header"
          >
            Last updated
          </span>
          <br />
          May 1, 2018
          <br />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-lg-up"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
        </div>
        <div
          className="m-more-projects-table__status"
        >
          <span
            className="m-more-projects-table__internal-header"
          >
            Status
          </span>
          <br />
          <i
            className="fa m-more-projects-table__status-icon m-more-projects-table__status-icon--contract-awarded"
          />
           
          <span
            className="m-more-projects-table__status-text m-more-projects-table__status-text--contract-awarded"
          >
            Contract awarded
          </span>
        </div>
      </div>
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
      <span
        aria-hidden="false"
        className="notranslate c-svg__icon m-more-projects-table__route-icon hidden-md-down"
        dangerouslySetInnerHTML={
          Object {
            "__html": "SVG",
          }
        }
      />
    </div>
  </td>
  <td
    className="m-more-projects-table__td m-more-projects-table__td-last-updated hidden-md-down"
  >
    May 1, 2018
  </td>
  <td
    className="m-more-projects-table__td m-more-projects-table__td-status hidden-md-down"
  >
    <span
      className="m-more-projects-table__status-wrapper"
    >
      <i
        className="fa m-more-projects-table__status-icon m-more-projects-table__status-icon--contract-awarded"
      />
       
      <span
        className="m-more-projects-table__status-text m-more-projects-table__status-text--contract-awarded"
      >
        Contract awarded
      </span>
    </span>
  </td>
</tr>
`;
